services/src/test/java/edu/unc/lib/dl/cdr/services/rest/RetrieveMODSIT.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.cdr.services.rest;
+
+import static edu.unc.lib.dl.acl.util.Permission.viewMetadata;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+
+import org.apache.tika.io.IOUtils;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.ContextHierarchy;
+import org.springframework.test.web.servlet.MvcResult;
+
+import edu.unc.lib.dl.acl.exception.AccessRestrictionException;
+import edu.unc.lib.dl.acl.util.AccessGroupSet;
+import edu.unc.lib.dl.cdr.services.rest.modify.AbstractAPIIT;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectFactory;
+import edu.unc.lib.dl.fcrepo4.WorkObject;
+import edu.unc.lib.dl.fedora.PID;
+
+/**
+ *
+ * @author harring
+ *
+ */
+@ContextHierarchy({
+    @ContextConfiguration("/spring-test/test-fedora-container.xml"),
+    @ContextConfiguration("/spring-test/cdr-client-container.xml"),
+    @ContextConfiguration("/retrieve-mods-it-servlet.xml")
+})
+public class RetrieveMODSIT extends AbstractAPIIT {
+
+    @Autowired
+    private RepositoryObjectFactory repositoryObjectFactory;
+
+    @Test
+    public void testRetrieveMODSFromWork() throws Exception {
+        doNothing().when(aclService).assertHasAccess(anyString(), any(PID.class), any(AccessGroupSet.class),
+                eq(viewMetadata));
+        PID workPid = makePid();
+        File modsFile = setupWorkWithMODS(workPid);
+
+        MvcResult result = mvc.perform(get("/description/" + workPid.getUUID()))
+                .andExpect(status().is2xxSuccessful())
+                .andReturn();
+
+        String responseBody = result.getResponse().getContentAsString();
+        String expectedResponseBody = IOUtils.toString(new FileInputStream(modsFile));
```

Comment:
You can use `FileUtils.readFileToString` directly to read the file

Reply:
nifty :-)